hs_collection_resource/views.py

```diff
@@ -0,0 +1,106 @@
+import json
+import logging
+
+from django.http import HttpResponseRedirect, HttpResponseNotFound, HttpResponse, JsonResponse
+
+from hs_core import hydroshare
+from hs_core.views import _set_resource_sharing_status
```

Comment:
@pkdash  After trying to change that hs_core func name (_set_resource_sharing_status) and moving it from __init__ to utils.py, I finally ended up with a different solution:

I slightly changed hs_core view func set_resource_flag() such that we can either call its endpoint api ( /res_id/set-resource-flag/) in post mode or directly call this view function with parameters mimicking a post request. In this way, I don't need to duplicate this piece of custom logic for collection in my view func update_collection().  Please let me know if it is good or not. Thanks

# modified hs_core view func set_resource_flag()
https://github.com/hydroshare/hydroshare/pull/1000/files#diff-1d9fbaa49f3ea3ec9956ffcd21ac7629R345

# modified my view func update_collection()
https://github.com/hydroshare/hydroshare/pull/1000/files#diff-99c8ca8edbd2d800a67be1aece7f4165R56